Sources/App/Controllers/HomekitAppController.swift

```diff
@@ -0,0 +1,149 @@
+//
+//  HomekitAppController.swift
+//  App
+//
+//  Created by Kim de Vos on 29/07/2018.
+//
+
+import Foundation
+import Vapor
+import HTTP
+import FluentPostgreSQL
+import Leaf
+import Fluent
+
+final class HomekitAppController {
+
+    init(router: Router) {
+        router.get("apps", use: apps)
+        router.get("apps", Int.parameter, use: app)
+        router.get("apps","contribute", use: contribute)
+        router.post("apps","contribute", use: submit)
+    }
+
+    func apps(_ req: Request) throws -> Future<View> {
+        let categories = try QueryHelper.categories(request: req)
+        let manufacturerCount = try QueryHelper.manufacturerCount(request: req)
+        let apps = try QueryHelper.apps(request: req)
+        let accessoryCount = try QueryHelper.accessoriesCount(request: req)
+        
+        return flatMap(to: View.self, categories, manufacturerCount, apps, accessoryCount) { categories, manufacturersCount, apps, accessoryCount in
+            let data = AppsResponse(apps: apps,
+                                    categories: categories,
+                                    accessoryCount: accessoryCount,
+                                    manufacturerCount: manufacturersCount)
+            let leaf = try req.view()
+            return leaf.render("Apps/apps", data)
+        }
+    }
+
+    func app(_ req: Request) throws -> Future<View> {
+        let paramId: Int = try req.parameters.next()
+
+        let categories = try QueryHelper.categories(request: req)
+        let manufacturerCount = try QueryHelper.manufacturerCount(request: req)
+        let app = try QueryHelper.app(request: req, id: paramId)
+        let accessoryCount = try QueryHelper.accessoriesCount(request: req)
+
+        return app.flatMap{ app in
+            guard let app = app else { throw Abort(.badRequest) }
+
+            return flatMap(to: View.self, categories, manufacturerCount, accessoryCount) { categories, manufacturersCount, accessoryCount in
+                let data = AppResponse(app: app,
+                                       categories: categories,
+                                       accessoryCount: accessoryCount,
+                                       manufacturerCount: manufacturersCount)
+                let leaf = try req.view()
+                return leaf.render("Apps/app", data)
+            }
+        }
+    }
+
+    func contribute(_ req: Request) throws -> Future<View> {
```

Comment:
@BalestraPatrick I was thinking of we should add contribution to the pages to `accessory/contribute` and `apps/contribute`

Be cause I think one big contribute page with a dynamic form for apps and accessories will be messy.

Reply:
What do you think?

Reply:
Sure, that sounds good in this case.